### PR TITLE

LPS-178022 Remove Stream and Optional usages in portal-search-web final

### PR DESCRIPTION
**Pull request by julschong**
@tinatian resubmitted from https://github.com/liferay-core-infra/liferay-portal/pull/4419 from our discussion

see commit comment for details.

other changes are just moving commited around for better organization

### CODE REVIEW

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/facet/display/context/builder/CustomFacetDisplayContextBuilder.java [hunk 1]
```diff
@@ -154,10 +153,11 @@ public CustomFacetDisplayContextBuilder setParameterValue(
 	}
 
 	public CustomFacetDisplayContextBuilder setParameterValues(
-		Optional<List<String>> parameterValuesOptional) {
+		List<String> parameterValues) {
```
Comment (julschong) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/facet/display/context/builder/CustomFacetDisplayContextBuilder.java:156]: Completely removed Optional usage here

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/facet/portlet/CustomFacetPortlet.java [hunk 2]
```diff
@@ -142,7 +142,11 @@ private CustomFacetDisplayContext _buildDisplayContext(
 		).setParameterName(
 			parameterName
 		).setParameterValues(
-			parameterValuesOptional
+			optional.map(
+				Arrays::asList
+			).orElse(
+				Collections.emptyList()
+			)
```
Comment (julschong) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/facet/portlet/CustomFacetPortlet.java:149]: use orElse here for usage

**Comment by catarineaguiar** [comment]
ci:test:search

**Comment by catarineaguiar** [comment]
@julschong possible regression:

```
     [exec] com.liferay.portal.search.elasticsearch7.internal.spellcheck.ElasticsearchSpellCheckTest > testSpellCheckMap FAILED
     [exec]     org.junit.ComparisonFailure: expected:<{lifera=[[liferay]]}> but was:<{lifera=[[]]}>
     [exec]         at org.junit.Assert.assertEquals(Assert.java:115)
     [exec]         at org.junit.Assert.assertEquals(Assert.java:144)
     [exec]         at com.liferay.portal.search.test.util.spellcheck.BaseSpellCheckTestCase.lambda$assertSpellCheckMap$1(BaseSpellCheckTestCase.java:169)
     [exec]         at com.liferay.portal.search.test.util.IdempotentRetryAssert.retryAssert(IdempotentRetryAssert.java:43)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-37/test-portal-acceptance-pullrequest(master)/11119/modules-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by julschong** [comment]
@catarineaguiar the test passed locally for me, would you please help me check?

**Comment by julschong** [comment]
ci:test:search

**Comment by julschong** [comment]
https://github.com/liferay-core-infra/liferay-portal/pull/4433#issuecomment-1533770320
@Kvale1733 Hi Kevin, could you please help me check on this?

**Comment by Kvale1733** [comment]
@julschong It's not a regression. Looks like a fluke since it passed on relevant and passes locally.

**Comment by tinatian** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/4458

**Pull request closed without merging**